### PR TITLE
LPS-199287 Merge File interface and impl into Util in kernel

### PR DESCRIPTION
**Pull request by julschong**
@shuyangzhou 
Although this FileUtil change stays in kernel, it still contains breaking change to release repo due to FileUtil.getFile() and setFile() being used.
I sent release code PR here: https://github.com/liferay/liferay-release-tool-ee/pull/23

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:relevant

File: modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/platform.bndrun [hunk 1]
```diff
@@ -15,7 +15,6 @@ $h1 Check GETTING_STARTED.markdown#platform.bndrun for more information.
 	osgi.service;objectClass:List<String>='com.liferay.portal.kernel.portlet.PortletPreferencesFactory',\
 	osgi.service;objectClass:List<String>='com.liferay.portal.kernel.portlet.PortletURLFactory',\
 	osgi.service;objectClass:List<String>='com.liferay.portal.kernel.service.persistence.LayoutSetPersistence',\
-	osgi.service;objectClass:List<String>='com.liferay.portal.kernel.util.File',\
```
Comment (julschong) [inline modules/sdk/project-templates/project-templates-workspace/src/main/resources/archetype-resources/platform.bndrun:18]: @shuyangzhou I am not very sure about this change. 
Do you know what this does?

**Comment by julschong** [comment]
fix sf

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
@julschong there are some conflicts, please resolve them. Thanks

**Comment by catarineaguiar** [comment]
@julschong btw, many tests failed due to StackOverflowError:

```
[beanshell]      [exec] com.liferay.user.service.test.UserLocalServiceTest > testGetCompanyUsers FAILED
[beanshell]      [exec]     java.lang.StackOverflowError
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
[beanshell]      [exec]     	at com.liferay.portal.kernel.util.FileUtil.getBytes(FileUtil.java:325)
```
```
    [junit] 2023-10-17 22:08:01.389 ERROR [main][ROOT:47] bundle com.liferay.portal.k8s.agent.impl:3.0.33 (19)[com.liferay.portal.k8s.agent.internal.model.listener.VirtualHostModelListener(14)] : The activate method has thrown an exception
    [junit] java.lang.StackOverflowError: null
    [junit] 	at com.liferay.portal.kernel.util.FileUtil.write(FileUtil.java:756) ~[portal-kernel.jar:?]
    [junit] 	at com.liferay.portal.kernel.util.FileUtil.write(FileUtil.java:756) ~[portal-kernel.jar:?]
    [junit] 	at com.liferay.portal.kernel.util.FileUtil.write(FileUtil.java:756) ~[portal-kernel.jar:?]
    [junit] 	at com.liferay.portal.kernel.util.FileUtil.write(FileUtil.java:756) ~[portal-kernel.jar:?]
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-10/test-1-34/test-portal-acceptance-pullrequest(master)/4830/modules-integration-mysql57-jdk8/4/12/jenkins-console.txt.gz?authuser=0)

**Comment by julschong** [comment]
rebased and resolved conflict

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
ci:close

**Comment by zhangyan0701** [comment]
> I tried just changing com.liferay.portal.kernel.util.File to com.liferay.portal.kernel.util.FileUtil in the target but that didn't work. @zhangyan0701 Since you created this method, maybe you know how to fix it?

@julschong has fixed this one in https://github.com/liferay-core-infra/liferay-portal/commit/861b68f23fdd0aad350ba530a7d23bead0685b1b. I have tried locally and the fix worked.

The clustering tests also passed in the new pull https://github.com/liferay-core-infra/liferay-portal/pull/5852.

**Pull request closed without merging**